00-basics/002-javascript/freeCodeCamp_03_17.js

```diff
@@ -0,0 +1,218 @@
+function myStr() {
+      let someAdjective = "awesome!";
+     let myStr = "Learning to code is ";
+    myStr += someAdjective;
+    console.log(myStr);
+}
+myStr();
+
+function lastNameLength() {
+     lastNameLength = 0;
+    let lastName = "Lovelace";
+    lastNameLength = lastName.length;
+    console.log(lastNameLength);
+}
+lastNameLength();
+
+function firstLetter() {
+    let firstLetterOfLastName = "";
+    let lastName = "Lovelace";
+    firstLetterOfLastName = lastName[0];
+    console.log(firstLetterOfLastName);
+}
+firstLetter();
+
+function undStr () {
+     myStr = "Jello World";
```

Comment:
You probably know this already, but this results in setting a global value `myStr`. This approach is sometimes used in frontend, and **never** in the backend.
`var` is global when declared outside of a function and function-scoped if declared within a function. `let` and `const` are always block-scoped.

Reply:
I'm confused with this place. If I declare let myStr = "Jello World", I get the message "Variable initializer is redundant"
If I do'nt declare, I leave myStr = "JelloWorld" - no message. I don’t understand this place.

Reply:
Considering that functions are also variables in JS, I think what is happening is that when you say `mystr`, you are actually referring to `function myStr()`. And when you add declaration operator, JS complains that you are trying to redeclare variable that previously was defined as function.

Reply:
Thanks, Igori!